source/azure_iot_hub_client.c

```diff
@@ -0,0 +1,582 @@
+}
+/*-----------------------------------------------------------*/
+
+void AzureIoTHubClient_Deinit( AzureIoTHubClient_t * pxAzureIoTHubClient )
+{
+    ( void ) pxAzureIoTHubClient;
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTHubClientResult_t AzureIoTHubClient_Connect( AzureIoTHubClient_t * pxAzureIoTHubClient,
+                                                     bool cleanSession,
+                                                     uint32_t ulTimeoutMilliseconds )
+{
+    AzureIoTMQTTConnectInfo_t xConnectInfo = { 0 };
+    AzureIoTHubClientResult_t xResult;
+    AzureIoTMQTTStatus_t xMQTTResult;
+    bool xSessionPresent;
+    uint32_t ulPasswordLength = 0;
+    size_t xMQTTUserNameLength;
+    az_result xEmbeddedResult;
+
+    if( pxAzureIoTHubClient == NULL )
+    {
+        AZLogError( ( "Failed to connect: Invalid argument." ) );
+        xResult = eAzureIoTHubClientInvalidArgument;
+    }
+    else
+    {
+        /* Use scratch buffer for username/password */
+        xConnectInfo.pUserName = ( const char * ) pxAzureIoTHubClient->_internal.pucAzureIoTHubClientWorkingBuffer;
+        xConnectInfo.pPassword = xConnectInfo.pUserName + azureiotUSERNAME_MAX;
+
+        if( az_result_failed( xEmbeddedResult = az_iot_hub_client_get_user_name( &pxAzureIoTHubClient->_internal.xAzureIoTHubClientCore,
+                                                                                 ( char * ) xConnectInfo.pUserName, azureiotUSERNAME_MAX,
+                                                                                 &xMQTTUserNameLength ) ) )
+        {
+            AZLogError( ( "Failed to get username: %08x.", xEmbeddedResult ) );
+            xResult = eAzureIoTHubClientFailed;
+        }
+        /* Check if token refresh is set, then generate password */
+        else if( ( pxAzureIoTHubClient->_internal.pxAzureIoTHubClientTokenRefresh ) &&
+                 ( pxAzureIoTHubClient->_internal.pxAzureIoTHubClientTokenRefresh( pxAzureIoTHubClient,
+                                                                                   pxAzureIoTHubClient->_internal.xAzureIoTHubClientTimeFunction() + azureiothubDEFAULT_TOKEN_TIMEOUT_IN_SEC,
+                                                                                   pxAzureIoTHubClient->_internal.pucAzureIoTHubClientSymmetricKey,
+                                                                                   pxAzureIoTHubClient->_internal.ulAzureIoTHubClientSymmetricKeyLength,
+                                                                                   ( uint8_t * ) xConnectInfo.pPassword, azureiotPASSWORD_MAX,
+                                                                                   &ulPasswordLength ) ) )
+        {
+            AZLogError( ( "Failed to generate auth token." ) );
+            xResult = eAzureIoTHubClientFailed;
+        }
+        else
+        {
+            xConnectInfo.cleanSession = cleanSession;
```

Comment:
>cleanSession [](start = 40, length = 12)

xConnection info might have changed its member name. Can you please check.

Reply:
yup will do